protocol/bc/transaction.go

```diff
@@ -267,6 +250,14 @@ func (tx *TxData) HashForSig(idx uint32) Hash {
 	return NewSigHasher(tx).Hash(idx)
 }
 
+func (tx *Tx) OutputID(outputIndex int) OutputID {
+	return ComputeOutputID(tx.Hash, uint32(outputIndex))
```

Comment:
This is an unsafe typecast. Why not require `outputIndex` to be a `uint32` on input to the function? (Here and below.)